core/iwasm/fast-jit/fe/jit_emit_numberic.c

```diff
@@ -472,16 +472,128 @@ jit_compile_op_i64_bitwise(JitCompContext *cc, IntBitwise bitwise_op)
     return false;
 }
 
+static bool
+compile_int_shift(JitCompContext *cc, IntShift shift_op, bool is_i32)
+{
+#if defined(BUILD_TARGET_X86_64) || defined(BUILD_TARGET_AMD_64)
+    JitReg ecx_hreg = jit_codegen_get_hreg_by_name("ecx");
+    JitReg rcx_hreg = jit_codegen_get_hreg_by_name("rcx");
+#endif
+    JitReg left, right, mod_right, res;
+
+    POP_INT(right);
+    POP_INT(left);
+
+    /* right modulo N */
+    if (jit_reg_is_const(right)) {
+        if (is_i32) {
+            int32 right_value = jit_cc_get_const_I32(cc, right);
+            right_value = right_value % 32;
+            if (0 == right_value) {
+                res = left;
+                goto shortcut;
+            }
+            else {
+                mod_right = NEW_CONST(I32, right_value);
+            }
+        }
+        else {
+            int64 right_value = jit_cc_get_const_I64(cc, right);
+            right_value = right_value % 64;
+            if (0 == right_value) {
+                res = left;
+                goto shortcut;
+            }
+            else {
+                mod_right = NEW_CONST(I64, right_value);
+            }
+        }
+    }
+    else {
+        if (is_i32) {
+            mod_right = jit_cc_new_reg_I32(cc);
+            GEN_INSN(REM_U, mod_right, right, NEW_CONST(I32, 32));
```

Comment:
An improvement: (right mod 32) equals to (right & 0x1F), do want want to change to `GEN_INSN(AND, mod_right, right, NEW_CONST(I32, 0x1F)`?

Reply:
Sure, `AND` is better